csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -1,303 +1,515 @@
-				{"New measure value from measure column": {"$ref": "#/$defs/measure" }}
-			]
-		},
-
-
-		"dataTypes": {
-			"type": "string",
-			"default": "decimal",
-			"enum": [
-				"anyURI",
-				"boolean",
-				"date",
-				"dateTime",
-				"dateTimeStamp",
-				"decimal",
-				"integer",
-				"long",
-				"int",
-				"short",
-				"nonNegativeInteger",
-				"positiveInteger",
-				"unsignedLong",
-				"unsignedInt",
-				"unsignedShort",
-				"nonPositiveInteger",
-				"negativeInteger",
-				"double",
-				"float",
-				"string",
-				"language",
-				"time"]
-		},
-
-		"existingDimension": {
-            "description": "An existing dimension",
-            "$ref": "#/$defs/dimension"
-        },
-
-        "newDimension": {
-            "description": "A new dimension",
-            "$ref": "#/$defs/dimension",
-            "properties": {
-				"codelist": {
-                    "description": "Source URI for the code-list for the dimension, e.g. 'http://example.org/code-list/{+some_column}'",
-                    "type": "string",
-                    "format": "uri"
+      "columnTypes": {
+        "attribute": {
+          "description": "An Attribute column",
+          "type": "object",
+          "required": [
+            "type"
+          ],
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "description": "The type of column",
+              "type": "string",
+              "const": "attribute"
+            },
+            "description": {
+              "description": "A description of the attribute",
+              "type": "string"
+            },
+            "from_existing": {
+              "description": "The URI of an existing attribute column",
+              "anyOf": [
+                {
+                  "description": "The relative file path to the attribute's local definition",
+                  "type": "string",
+                  "format": "uri"
+                },
+                {
+                  "description": "The URI of the existing attribute",
+                  "$ref": "http://gss-cogs.github.io/family-schemas/external.json#/definitions/attr_ext_uri"
+                }
+              ]
+            },
+            "label": {
+              "description": "The replacement label for the attribute",
+              "type": "string"
+            },
+            "definition_uri": {
+              "description": "The URI that defines the attribute source",
+              "anyOf": [
+                {
+                  "description": "The relative file path to the code-list's local definition",
+                  "type": "string",
+                  "format": "uri"
+                },
+                {
+                  "description": "The URI of the codelist associated with the dimension.",
+                  "$ref": "http://gss-cogs.github.io/family-schemas/external.json#/definitions/attr_ext_uri"
```

Comment:
Definition URIs link to things like PDF documents a user wants to publish and shouldn't provide any auto-complete help like you've got here. These URIs aren't suitable since they're not helpful to a human.

Reply:
updated.